packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm

```diff
@@ -476,7 +476,11 @@ - (void)_loadScriptFromSource:(RCTSource *)source
   auto script = std::make_unique<NSDataBigString>(source.data);
   const auto *url = deriveSourceURL(source.url).UTF8String;
   _reactInstance->loadScript(std::move(script), url);
-  [[NSNotificationCenter defaultCenter] postNotificationName:@"RCTInstanceDidLoadBundle" object:nil];
+  [self callFunctionOnBufferedRuntimeExecutor:[](facebook::jsi::Runtime &_) {
```

Comment:
This jumps to a different thread and then we jumps back to the original one. I'm not super excited about this: what do you think about changing the `ReactInstance::loadScript` method to take an extra, optional parameter (`nullptr` as default value) which is a callback and invoke the callback when `loadScript` finishes?

It should prevent us a thread jump and ensure that we are actually posting the notification when the script has been loaded.

We can check [here](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/runtime/ReactInstance.cpp#L264) if the callback is set and invoke it.

Reply:
@cipolleschi Done.